types/src/v0/impls/stake_table.rs

```diff
@@ -216,17 +223,57 @@ impl StaticCommittee {
             indexed_stake_table,
             indexed_da_stake_table,
             _epoch_size: epoch_size,
-            _provider: instance_state.l1_client.clone(),
+            l1_client: instance_state.l1_client.clone(),
             _contract_address: instance_state.chain_config.stake_table_contract,
         }
     }
+
+    async fn update_loop(committee: Arc<RwLock<StaticCommittee>>) -> impl Future<Output = ()> {
+        let l1_client = { committee.read().await.l1_client.clone() };
+
+        let retry_delay = l1_client.retry_delay;
+
+        let span = tracing::warn_span!("updating committee stake tables");
+        async move {
+            loop {
+                let mut events = l1_client.receiver.activate_cloned();
+                while let Some(event) = events.next().await {
+                    let L1Event::NewEpoch {
+                        epoch,
+                        l1_block_number,
+                    } = event
+                    else {
+                        continue;
+                    };
+
+                    loop {
+                        {
+                            let mut committee_lock = committee.write().await;
+                            if let Err(err) =
+                                committee_lock.update_stake_table(l1_block_number).await
```

Comment:
If we do need a lock I think it will be a lot safer to take it out only when we change the in memory stake table but not while we fetch it from the RPC.